src/WebJobs.Script/Binding/DocumentDBScriptBindingProvider.cs

```diff
@@ -89,25 +104,54 @@ public override Collection<Attribute> GetAttributes()
 
                 string databaseName = Context.GetMetadataValue<string>("databaseName");
                 string collectionName = Context.GetMetadataValue<string>("collectionName");
+                string connection = Context.GetMetadataValue<string>("connection");
 
-                DocumentDBAttribute attribute = null;
-                if (!string.IsNullOrEmpty(databaseName) || !string.IsNullOrEmpty(collectionName))
+                if (Context.IsTrigger)
                 {
-                    attribute = new DocumentDBAttribute(databaseName, collectionName);
+                    DocumentDBTriggerAttribute attributeTrigger = new DocumentDBTriggerAttribute(databaseName, collectionName);
+                    attributeTrigger.ConnectionString = connection;
+                    string leaseCollectionName = Context.GetMetadataValue<string>("leaseCollectionName");
+                    string leaseDatabaseName = Context.GetMetadataValue<string>("leaseDatabaseName");
+                    string leaseConnectionString = Context.GetMetadataValue<string>("leaseConnection");
+
+                    if (!string.IsNullOrEmpty(leaseCollectionName))
+                    {
+                        attributeTrigger.LeaseCollectionName = leaseCollectionName;
```

Comment:
@mamaso By default, if no lease properties are set, the Lease database is the same as the `databaseName` attribute, the Lease colletion defaults to "leases" and the connection string is the same as the `connection` attribute. So it uses the same service, same database but different collection. We are also checking that an [infinite loop is not created](https://github.com/ealsur/azure-webjobs-sdk-extensions/blob/dev/src/WebJobs.Extensions.DocumentDB/Trigger/DocumentDBTriggerAttributeBindingProvider.cs#L77) where both collections are the same.

I will rework the properties of the leases in a nested object in the next commit 😄